main.tf

```diff
@@ -433,6 +442,38 @@ resource "null_resource" "deliver_data_bags" {
   }
 }
 
+data "archive_file" "support" {
+  count       = (local._support_supplied == false) || (var.skip == true) ? 0 : 1
+  type        = "zip"
+  source_file = local._support_supplied_is_file ? local.supplied_support : ""
+  source_dir  = local._support_supplied_is_dir ? local._support_supplied_dirname : ""
```

Comment:
if this doesn't work, you're going to have to make two separate data resources, one which has count = 1 for dir and the other for a file